tests/testsuite/doc.rs

```diff
@@ -1306,6 +1329,210 @@ fn doc_workspace_open_help_message() {
         .run();
 }
 
+#[cargo_test(nightly, reason = "-Zper-package-target is unstable")]
```

Comment:
I don't think this is necessary.  Limiting to nightly is only necessary if running something in rustc or rustdoc that needs nightly.  per-package-target is a purely a cargo thing, which is handled by `masquerade_as_nightly_cargo`.

```suggestion
#[cargo_test]
```